### ISSUE TITLE
Documentation for custom function

### ISSUE THREAD

**Issue by cisaacson** [pola-rs/polars#8035]
### Problem description

`polars` is working very well during our testing. A custom function for a map in python has this documentation:

https://pola-rs.github.io/polars/py-polars/html/reference/expressions/api/polars.map.html

What is missing is an example of a python function that works with polars types with a bit more detail. The python Series API is very easy, but there was no example that showed this. I am not sure this is the most efficient, but if the function needs to complex work on each individual value in the Series, this approach works:

```
def test_func2(a: pl.Series, b: pl.Series) -> pl.Series:
  result_list: list[int] = []
  for i, a in enumerate(a):
    b = b[i]
    x = do_something_complex(a, b)
    result_list.append(x)

  result_series = pl.Series(result_list)

  return result_series
```

**Comment by StefanBRas** [comment]
You probably want [`polars.Expr.apply`](https://pola-rs.github.io/polars/py-polars/html/reference/expressions/api/polars.Expr.apply.html).

This is also documented in the [Custom functions](https://pola-rs.github.io/polars-book/user-guide/dsl/custom_functions.html) section of the user guide.

**Comment by cisaacson** [comment]
Yes, I read about this but also in the doc is how much more efficient map is (since you can send all the data in one call). I think that is best for my requirement, let me know if otherwise.

My main point was there was no documentation of how to work with a Series object in Python. It has a nice and clean API, would be great to have examples that show that.

**Comment by zundertj** [comment]
@cisaacson : a PR to add example(s) to the docstring are always welcome.
